common/lib/xmodule/xmodule/tests/test_course_module.py

```diff
@@ -277,3 +278,71 @@ class DiscussionTopicsTestCase(unittest.TestCase):
     def test_default_discussion_topics(self):
         d = get_dummy_course('2012-12-02T12:00')
         self.assertEqual({'General': {'id': 'i4x-test_org-test_course-course-test'}}, d.discussion_topics)
+
+
+class TeamsConfigurationTestCase(unittest.TestCase):
+    """
+    Tests for the configuration of teams and the helper methods for accessing them.
+    """
+
+    def setUp(self):
+        super(TeamsConfigurationTestCase, self).setUp()
+        self.course = get_dummy_course('2012-12-02T12:00')
+        self.course.teams_configuration = dict()
+        self.count = itertools.count()
+
+    def add_team_configuration(self, max_team_size=3, topics=None):
+        """ Add a team configuration to the course. """
+        if topics is None:
+            topics = []
+        self.course.teams_configuration = {"max_team_size": max_team_size, "topics": topics}
+
+    def make_topic(self):
+        """ Make a sample topic dictionary. """
+        next_num = self.count.next()
+        topic_id = "topic_id_{}".format(next_num)
+        display_name = "Display Name {}".format(next_num)
+        description = "Description {}".format(next_num)
+        return {"display_name": display_name, "description": description, "id": topic_id}
+
+    def test_teams_enabled_new_course(self):
+        # Make sure we can detect when no teams exist.
+        self.assertFalse(self.course.teams_enabled)
+
+        # add topics
+        self.add_team_configuration(max_team_size=4, topics=[self.make_topic()])
+        self.assertTrue(self.course.teams_enabled)
+
+        # remove them again
+        self.add_team_configuration(max_team_size=4, topics=[])
+        self.assertFalse(self.course.teams_enabled)
+
+    def test_teams_enabled_max_size_only(self):
+        self.add_team_configuration(max_team_size=4)
+        self.assertFalse(self.course.teams_enabled)
+
+    def test_teams_enabled_with_topics(self):
```

Comment:
I think this just duplicates test_teams_enabled_new_course.

It might be nice though to have a test where only topics are specified, not max_team_size (to make it clear that max_team_size is not required).